### PR TITLE
docs: rfc-014 engine conformance test suite

### PR DESCRIPTION
**Pull request by ehotting**
## Summary

- Language-agnostic JSON conformance test suite per schema version
- Conformance levels: Core, Cross-law, IoC, Temporal, Advanced
- Test format allows single and multi-regulation scenarios, error assertions, and optional trace assertions
- Provenance conformance tests verify Execution Receipt fields (RFC-013)
- Initial tests derived from existing BDD scenarios, then extended with edge cases

Depends on: RFC-013 (execution provenance, Execution Receipt format)

## Test plan

- [ ] Review RFC content with team
- [ ] Verify operation coverage matches current schema v0.5.1
- [ ] Verify conformance levels align with RFC-009 multi-org requirements

### CODE REVIEW

File: packages/engine/src/types.rs [hunk 1]
```diff
@@ -550,6 +550,44 @@ pub enum Operation {
...
+        Operation::Min,
+        Operation::And,
+        Operation::Or,
+        Operation::Not,
+        Operation::If,
+        Operation::In,
+        Operation::List,
+        Operation::Age,
+        Operation::DateAdd,
+        Operation::Date,
```
Comment (ehotting) [inline packages/engine/src/types.rs]: Fixed in becfd0e. Replaced the hardcoded count with `Operation::variant_count()` which uses an exhaustive match internally — adding a new enum variant without updating the match is now a compile error. The unit test then asserts `SCHEMA_OPERATIONS.len() + COMPAT_ALIASES.len() == variant_count()`.
